Makefile

```diff
@@ -1,96 +1,79 @@
-OUTPUT := .output
-TARGET := event_monitor_ebpf
+.PHONY: all
+all: build
+
+# environment:
+ARCH := $(shell uname -m)
 KERN_RELEASE := $(shell uname -r)
 KERN_SRC := /lib/modules/$(KERN_RELEASE)/build
-ARCH := $(shell uname -m | sed 's/x86_64/x86/')
-LLC ?= llc
-CLANG ?= clang
-LLVM_STRIP ?= llvm-strip
-BPF_C = tracee/${TARGET:=.c}
-BPF_OBJ = ${OUTPUT}/${TARGET:=.o}
-LIBBPF_SRC = $(abspath 3rdparty/libbpf/src)
-LIBBPF_OBJ = $(abspath $(OUTPUT)/libbpf.a)
-CFLAGS ?= -I$(OUTPUT)/usr/include/
+# inputs and outputs:
+OUT_DIR := dist
+BPF_SRC := tracee/tracee_ebpf.c 
+BPF_OBJ := $(OUT_DIR)/tracee_ebpf.o
+GO_SRC := $(shell find . -type f -name '*.go' ! -name '*_test.go')
+OUT_BIN := $(OUT_DIR)/tracee
+LIBBPF_SRC := 3rdparty/libbpf/src
+LIBBPF_HEADERS := $(OUT_DIR)/libbpf/usr/include
+LIBBPF_OBJ := $(OUT_DIR)/libbpf/libbpf.a
+OUT_DOCKER := tracee
+# tools:
+LLC := llc
+CLANG := clang
+LLVM_STRIP := llvm-strip
+# DOCKER := docker
+# GORELEASER := goreleaser
 
-SRC = $(shell find . -type f -name '*.go' ! -name '*_test.go' )
-ebpfProgramBase64 = $(shell base64 -w 0 tracee/event_monitor_ebpf.c)
+$(OUT_DIR):
+	mkdir -p $@
 
 .PHONY: build
-build: llvm-check $(BPF_OBJ) dist/tracee
-
-.PHONY: build-docker
-build-docker: clean
-	img=$$(docker build --target builder -q  .) && \
-	cnt=$$(docker create $$img) && \
-	docker cp $$cnt:/tracee/dist - | tar -xf - ; \
-	docker rm $$cnt ; docker rmi $$img
-
-dist/tracee: $(SRC) tracee/event_monitor_ebpf.c
-	GOOS=linux go build -v -o dist/tracee -ldflags "-X github.com/aquasecurity/tracee/tracee.ebpfProgramBase64Injected=$(ebpfProgramBase64)"
+build: $(OUT_BIN)
 
-.PHONY: test
-test:
-	go test -v ./...
-
-.PHONY: clean
-clean:
-	rm -rf dist || true
-	cd $(LIBBPF_SRC) && $(MAKE) clean;
-	rm -f $(BPF_OBJ)
-	rm -f *.ll
-	rm -rf .output
+$(OUT_BIN): $(BPF_OBJ) $(LIBBPF_OBJ) $(GO_SRC) | $(OUT_DIR)
+	GOOS=linux GOARCH=$(ARCH:x86_64=amd64) \
+		CGO_CFLAGS="-I $(abspath $(LIBBPF_HEADERS))/bpf" \
+		CGO_LDFLAGS="$(abspath $(LIBBPF_OBJ))" \
+		go build -v -o $(OUT_BIN) \
+		-ldflags "-X github.com/aquasecurity/tracee/tracee.ebpfProgramB64Injected=$$(base64 -w 0 $(BPF_SRC))"
 
-imageName ?= tracee
-.PHONY: docker
-docker:
-	docker build -t $(imageName) .
+$(OUT_BIN)_slim: $(BPF_OBJ) $(LIBBPF_OBJ) $(GO_SRC) | $(OUT_DIR)
+	GOOS=linux GOARCH=$(ARCH:x86_64=amd64) go build -v -o $(OUT_BIN)
```

Comment:
Regarding ARCH substitution for x86 - Maybe do the same for ARM as well?
uname -m for arm64 gives aarch64, while go uses arm64 to cross compile

Reply:
do we support ARM? :)

Reply:
Like I said in the other comment:
"Let's use this opportunity to fix this for arm as well."
One day we will support it :-)

Reply:
it feels weird having special build mode for arm when we don't intend to support it soon.. people might get the wrong impression. maybe it's better to add this to the makefile in the same pr that adds arm support no?

Reply:
Sure, no problem